base/common/python/pki/nssdb.py

```diff
@@ -815,6 +817,54 @@ def create_request(
         finally:
             shutil.rmtree(tmpdir)
 
+    def create_request_with_wrap_key(
+            self,
+            subject_dn,
+            request_file,
+            token=None,
+            key_type='rsa',
+            key_size=None):
+
+        """
+        Generate a CSR for certs that need to perform key wrapping/unwrapping
+        on an HSM (dependent on make/model). Only RSA key type is supported.
+        """
+
+        tmpdir = tempfile.mkdtemp()
+        try:
+
+            cmd = [
+                'PKCS10Client',
+                '-d', self.directory
+            ]
+
+            token = self.get_effective_token(token)
+            password_file = self.get_password_file(tmpdir, token)
+
+            if token:
+                cmd.extend(['-h', token])
+
+            if password_file:
+                with open(password_file, 'r') as f:
+                    password = f.read()
+                cmd.extend(['-p', password])
```

Comment:
Does `PKCS10Client` have any option to specify a password file? Usually it's not recommended to specify a password as a command line parameter since it could expose the password in the logs.